diff --git a/iokit/Kernel/x86_64/IOSharedLock.s b/iokit/Kernel/x86_64/IOSharedLock.s
index d5e5eca..b6e18b6 100644
--- a/iokit/Kernel/x86_64/IOSharedLock.s
+++ b/iokit/Kernel/x86_64/IOSharedLock.s
@@ -58,3 +58,14 @@ LEAF(_ev_try_lock, 0)
 	setz		%dl
 	movzbl		%dl, %eax
 END(_OSSpinLockTry)
+
+LEAF(_OSSpinLockLock, 0)
+        xor %eax, %eax
+        orl $-1, %edx
+        lock
+        cmpxchgl %edx, (%rdi)
+        setz %dl
+        movzbl %dl, %eax
+        cmp %eax, 0
+        je _OSSpinLockLock
+END(_OSSpinLockLock)
