diff --git a/osfmk/x86_64/idt64.s b/osfmk/x86_64/idt64.s
index 13f3f02..615696a 100644
--- a/osfmk/x86_64/idt64.s
+++ b/osfmk/x86_64/idt64.s
@@ -455,8 +455,8 @@ L_dispatch_64bit:
 	/*
 	 * Save segment regs - for completeness since theyre not used.
 	 */
-	movl	%fs, R64_FS(%r15)
-	movl	%gs, R64_GS(%r15)
+	mov	%fs, R64_FS(%r15)
+	mov	%gs, R64_GS(%r15)
 
 	/* Save general-purpose registers */
 	mov	%rax, R64_RAX(%r15)
@@ -509,10 +509,10 @@ L_dispatch_U32: /* 32-bit user task */
 	/*
 	 * Save segment regs
 	 */
-	movl	%ds, R32_DS(%r15)
-	movl	%es, R32_ES(%r15)
-	movl	%fs, R32_FS(%r15)
-	movl	%gs, R32_GS(%r15)
+	mov	%ds, R32_DS(%r15)
+	mov	%es, R32_ES(%r15)
+	mov	%fs, R32_FS(%r15)
+	mov	%gs, R32_GS(%r15)
 
 	/*
 	 * Save general 32-bit registers
