diff --git a/libsyscall/Libsyscall.xcconfig b/libsyscall/Libsyscall.xcconfig
index 1055719..92392c9 100644
--- a/libsyscall/Libsyscall.xcconfig
+++ b/libsyscall/Libsyscall.xcconfig
@@ -22,13 +22,12 @@ OTHER_CFLAGS[sdk=tvos*] = $(inherited) -DNO_SYSCALL_LEGACY
 OTHER_CFLAGS[sdk=appletvos*] = $(inherited) -DNO_SYSCALL_LEGACY
 OTHER_CFLAGS[sdk=bridgeos*] = $(inherited) -DNO_SYSCALL_LEGACY
 GCC_PREPROCESSOR_DEFINITIONS = CF_OPEN_SOURCE CF_EXCLUDE_CSTD_HEADERS DEBUG _FORTIFY_SOURCE=0
-HEADER_SEARCH_PATHS = $(PROJECT_DIR)/mach $(PROJECT_DIR)/os $(PROJECT_DIR)/wrappers $(PROJECT_DIR)/wrappers/string $(PROJECT_DIR)/wrappers/libproc $(PROJECT_DIR)/wrappers/libproc/spawn $(BUILT_PRODUCTS_DIR)/internal_hdr/include $(BUILT_PRODUCTS_DIR)/mig_hdr/local/include $(BUILT_PRODUCTS_DIR)/mig_hdr/include $(SDKROOT)/System/Library/Frameworks/System.framework/PrivateHeaders
+HEADER_SEARCH_PATHS = $(PROJECT_DIR)/mach $(PROJECT_DIR)/os $(PROJECT_DIR)/wrappers $(PROJECT_DIR)/wrappers/string $(PROJECT_DIR)/wrappers/libproc $(PROJECT_DIR)/wrappers/libproc/spawn $(BUILT_PRODUCTS_DIR)/internal_hdr/include $(BUILT_PRODUCTS_DIR)/mig_hdr/local/include $(BUILT_PRODUCTS_DIR)/mig_hdr/include $(DEPENDENCIES_DIR)/System/Library/Frameworks/System.framework/Versions/B/PrivateHeaders
 WARNING_CFLAGS = -Wmost
-GCC_TREAT_WARNINGS_AS_ERRORS = YES
 GCC_WARN_ABOUT_MISSING_NEWLINE = YES
 CODE_SIGN_IDENTITY = -
 DYLIB_CURRENT_VERSION = $(RC_ProjectSourceVersion)
-DYLIB_LDFLAGS = -umbrella System -all_load -lCrashReporterClient
+DYLIB_LDFLAGS = -umbrella System -all_load
 DYLIB_LDFLAGS[sdk=iphoneos*] = $(inherited) -Wl,-sectalign,__DATA,__data,1000
 DYLIB_LDFLAGS[sdk=watchos*] = $(inherited) -Wl,-sectalign,__DATA,__data,1000
 DYLIB_LDFLAGS[sdk=tvos*] = $(inherited) -Wl,-sectalign,__DATA,__data,1000
diff --git a/libsyscall/Libsyscall.xcodeproj/project.pbxproj b/libsyscall/Libsyscall.xcodeproj/project.pbxproj
index 7dafe04..7e1ad07 100644
--- a/libsyscall/Libsyscall.xcodeproj/project.pbxproj
+++ b/libsyscall/Libsyscall.xcodeproj/project.pbxproj
@@ -134,8 +134,8 @@
 		A59CB95616669EFB00B064B3 /* stack_logging_internal.h in Headers */ = {isa = PBXBuildFile; fileRef = A59CB95516669DB700B064B3 /* stack_logging_internal.h */; };
 		A59CB9581666A1A200B064B3 /* munmap.c in Sources */ = {isa = PBXBuildFile; fileRef = A59CB9571666A1A200B064B3 /* munmap.c */; };
 		BA0D9FB1199031AD007E8A73 /* kdebug_trace.c in Sources */ = {isa = PBXBuildFile; fileRef = BA0D9FB0199031AD007E8A73 /* kdebug_trace.c */; };
-		BA4414AA18336A5F00AAE813 /* mach in CopyFiles */ = {isa = PBXBuildFile; fileRef = BA4414A51833697C00AAE813 /* mach */; };
-		BA4414AB18336A6400AAE813 /* servers in CopyFiles */ = {isa = PBXBuildFile; fileRef = BA4414A6183369A100AAE813 /* servers */; };
+		BA4414AA18336A5F00AAE813 /* mach in Copy Files */ = {isa = PBXBuildFile; fileRef = BA4414A51833697C00AAE813 /* mach */; };
+		BA4414AB18336A6400AAE813 /* servers in Copy Files */ = {isa = PBXBuildFile; fileRef = BA4414A6183369A100AAE813 /* servers */; };
 		BA4414AD18336A9300AAE813 /* mach in CopyFiles */ = {isa = PBXBuildFile; fileRef = BA4414A7183369C100AAE813 /* mach */; };
 		BA4414B518336E3600AAE813 /* mach in Copy Files */ = {isa = PBXBuildFile; fileRef = BA4414A51833697C00AAE813 /* mach */; };
 		BA4414B618336E3A00AAE813 /* servers in Copy Files */ = {isa = PBXBuildFile; fileRef = BA4414A6183369A100AAE813 /* servers */; };
@@ -229,7 +229,7 @@
 		C9D9BD57114B00600000D8B9 /* task.defs in Sources */ = {isa = PBXBuildFile; fileRef = C9D9BD0F114B00600000D8B9 /* task.defs */; };
 		C9D9BD58114B00600000D8B9 /* thread_act.defs in Sources */ = {isa = PBXBuildFile; fileRef = C9D9BD10114B00600000D8B9 /* thread_act.defs */; };
 		C9D9BD59114B00600000D8B9 /* vm_map.defs in Sources */ = {isa = PBXBuildFile; fileRef = C9D9BD11114B00600000D8B9 /* vm_map.defs */; };
-		C9FD8508166D6BD400963B73 /* tsd.h in CopyFiles */ = {isa = PBXBuildFile; fileRef = C9EE57F51669673D00337E4B /* tsd.h */; };
+		C9FD8508166D6BD400963B73 /* tsd.h in Copy Files */ = {isa = PBXBuildFile; fileRef = C9EE57F51669673D00337E4B /* tsd.h */; };
 		E214BDC81C2E358300CEE8A3 /* clonefile.c in Sources */ = {isa = PBXBuildFile; fileRef = E214BDC71C2E34E200CEE8A3 /* clonefile.c */; };
 		E2A0F3341C3B17D100A11F8A /* fs_snapshot.c in Sources */ = {isa = PBXBuildFile; fileRef = E2A0F3331C3B17D100A11F8A /* fs_snapshot.c */; };
 		E4216C311822D404006F2632 /* mach_voucher.defs in Sources */ = {isa = PBXBuildFile; fileRef = E4216C301822D404006F2632 /* mach_voucher.defs */; };
@@ -297,15 +297,16 @@
 /* End PBXContainerItemProxy section */
 
 /* Begin PBXCopyFilesBuildPhase section */
-		BA4414A818336A1300AAE813 /* CopyFiles */ = {
+		BA4414A818336A1300AAE813 /* Copy Files */ = {
 			isa = PBXCopyFilesBuildPhase;
 			buildActionMask = 8;
 			dstPath = "$(PUBLIC_HEADERS_FOLDER_PATH)";
 			dstSubfolderSpec = 0;
 			files = (
-				BA4414AA18336A5F00AAE813 /* mach in CopyFiles */,
-				BA4414AB18336A6400AAE813 /* servers in CopyFiles */,
+				BA4414AA18336A5F00AAE813 /* mach in Copy Files */,
+				BA4414AB18336A6400AAE813 /* servers in Copy Files */,
 			);
+			name = "Copy Files";
 			runOnlyForDeploymentPostprocessing = 1;
 		};
 		BA4414AC18336A7700AAE813 /* CopyFiles */ = {
@@ -371,14 +372,15 @@
 			);
 			runOnlyForDeploymentPostprocessing = 1;
 		};
-		C63F480B1654203800A1F78F /* CopyFiles */ = {
+		C63F480B1654203800A1F78F /* Copy Files */ = {
 			isa = PBXCopyFilesBuildPhase;
 			buildActionMask = 8;
 			dstPath = "$(OS_PRIVATE_HEADERS_FOLDER_PATH)";
 			dstSubfolderSpec = 0;
 			files = (
-				C9FD8508166D6BD400963B73 /* tsd.h in CopyFiles */,
+				C9FD8508166D6BD400963B73 /* tsd.h in Copy Files */,
 			);
+			name = "Copy Files";
 			runOnlyForDeploymentPostprocessing = 1;
 		};
 		C6D3EFCA16542C510052CF30 /* CopyFiles */ = {
@@ -1140,8 +1142,8 @@
 			buildConfigurationList = 1DEB914A08733D8E0010E9CD /* Build configuration list for PBXNativeTarget "Libsyscall_static" */;
 			buildPhases = (
 				D2AAC0600554660B00DB518D /* Headers */,
-				C63F480B1654203800A1F78F /* CopyFiles */,
-				BA4414A818336A1300AAE813 /* CopyFiles */,
+				C63F480B1654203800A1F78F /* Copy Files */,
+				BA4414A818336A1300AAE813 /* Copy Files */,
 				BA4414AC18336A7700AAE813 /* CopyFiles */,
 				D2AAC0610554660B00DB518D /* Sources */,
 				D289988505E68E00004EDB86 /* Frameworks */,
diff --git a/libsyscall/mach/host.c b/libsyscall/mach/host.c
index 651d148..b5b6c4b 100644
--- a/libsyscall/mach/host.c
+++ b/libsyscall/mach/host.c
@@ -27,7 +27,7 @@
  */
 
 #include <TargetConditionals.h>
-#include <machine/cpu_capabilities.h>
+#include <i386/cpu_capabilities.h>
 #include <mach/kern_return.h>
 #include <mach/mach_host.h>
 #include <mach/host_priv.h>
diff --git a/libsyscall/mach/mach_init.c b/libsyscall/mach/mach_init.c
index 338f7c9..39c6622 100644
--- a/libsyscall/mach/mach_init.c
+++ b/libsyscall/mach/mach_init.c
@@ -58,7 +58,7 @@
 #include <mach/mach_host.h>
 #include <mach/mach_init.h>
 #include <mach/vm_param.h>
-#include <machine/cpu_capabilities.h>
+#include <i386/cpu_capabilities.h>
 #include <stdbool.h>
 #include "externs.h"
 
diff --git a/libsyscall/wrappers/__commpage_gettimeofday.c b/libsyscall/wrappers/__commpage_gettimeofday.c
index 0ebfc53..97033d9 100644
--- a/libsyscall/wrappers/__commpage_gettimeofday.c
+++ b/libsyscall/wrappers/__commpage_gettimeofday.c
@@ -23,7 +23,7 @@
 
 #include <sys/time.h>
 #include <mach/mach_time.h>
-#include <machine/cpu_capabilities.h>
+#include <i386/cpu_capabilities.h>
 #include <os/overflow.h>
 #include <kern/arithmetic_128.h>
 
diff --git a/libsyscall/wrappers/__get_cpu_capabilities.s b/libsyscall/wrappers/__get_cpu_capabilities.s
index 86b0ee2..bf101ef 100644
--- a/libsyscall/wrappers/__get_cpu_capabilities.s
+++ b/libsyscall/wrappers/__get_cpu_capabilities.s
@@ -24,7 +24,7 @@
 /* Get the cpu_capabilities bit vector out of the comm page */
 
 #define	__APPLE_API_PRIVATE
-#include <machine/cpu_capabilities.h>
+#include <i386/cpu_capabilities.h>
 #undef	__APPLE_API_PRIVATE
 
 #if defined(__x86_64__)
diff --git a/libsyscall/wrappers/init_cpu_capabilities.c b/libsyscall/wrappers/init_cpu_capabilities.c
index 8414c1d..ad57a11 100644
--- a/libsyscall/wrappers/init_cpu_capabilities.c
+++ b/libsyscall/wrappers/init_cpu_capabilities.c
@@ -22,7 +22,7 @@
  */
 
 #define	__APPLE_API_PRIVATE
-#include <machine/cpu_capabilities.h>
+#include <i386/cpu_capabilities.h>
 #undef	__APPLE_API_PRIVATE
 
 #if defined(__i386__) || defined(__x86_64__)
diff --git a/libsyscall/wrappers/kdebug_trace.c b/libsyscall/wrappers/kdebug_trace.c
index 8234f45..74ae949 100644
--- a/libsyscall/wrappers/kdebug_trace.c
+++ b/libsyscall/wrappers/kdebug_trace.c
@@ -25,7 +25,7 @@
 #include <stdbool.h>
 #include <stdlib.h>
 #include <stdatomic.h>
-#include <machine/cpu_capabilities.h>
+#include <i386/cpu_capabilities.h>
 #include <sys/kdebug.h>
 #include <sys/kdebug_signpost.h>
 #include <sys/errno.h>
diff --git a/libsyscall/wrappers/mach_absolute_time.s b/libsyscall/wrappers/mach_absolute_time.s
index 2c637bf..f5061e2 100644
--- a/libsyscall/wrappers/mach_absolute_time.s
+++ b/libsyscall/wrappers/mach_absolute_time.s
@@ -27,7 +27,7 @@
  */
 
 #include <sys/appleapiopts.h>
-#include <machine/cpu_capabilities.h>
+#include <i386/cpu_capabilities.h>
 
 #if defined(__i386__)
 
diff --git a/libsyscall/wrappers/mach_approximate_time.c b/libsyscall/wrappers/mach_approximate_time.c
index 1e0a4e4..c2c271c 100644
--- a/libsyscall/wrappers/mach_approximate_time.c
+++ b/libsyscall/wrappers/mach_approximate_time.c
@@ -21,7 +21,7 @@
  * @APPLE_LICENSE_HEADER_END@
  */
 #include <sys/types.h>
-#include <machine/cpu_capabilities.h>
+#include <i386/cpu_capabilities.h>
 
 extern uint64_t mach_absolute_time(void);
 
diff --git a/libsyscall/wrappers/mach_approximate_time.s b/libsyscall/wrappers/mach_approximate_time.s
index f1e6ed8..ac4724d 100644
--- a/libsyscall/wrappers/mach_approximate_time.s
+++ b/libsyscall/wrappers/mach_approximate_time.s
@@ -27,7 +27,7 @@
  */
 
 #include <sys/appleapiopts.h>
-#include <machine/cpu_capabilities.h>
+#include <i386/cpu_capabilities.h>
 
 #if defined(__arm__)
 
diff --git a/libsyscall/wrappers/mach_boottime.c b/libsyscall/wrappers/mach_boottime.c
index 23b22f9..3ecdd89 100644
--- a/libsyscall/wrappers/mach_boottime.c
+++ b/libsyscall/wrappers/mach_boottime.c
@@ -20,7 +20,7 @@
  * 
  * @APPLE_LICENSE_HEADER_END@
  */
-#include <machine/cpu_capabilities.h>
+#include <i386/cpu_capabilities.h>
 
 uint64_t
 mach_boottime_usec(void)
diff --git a/libsyscall/wrappers/mach_continuous_time.c b/libsyscall/wrappers/mach_continuous_time.c
index 61b996d..22f7c22 100644
--- a/libsyscall/wrappers/mach_continuous_time.c
+++ b/libsyscall/wrappers/mach_continuous_time.c
@@ -23,7 +23,7 @@
 #include <stddef.h>
 #include <stdbool.h>
 #include <sys/types.h>
-#include <machine/cpu_capabilities.h>
+#include <i386/cpu_capabilities.h>
 #include <mach/mach_time.h>
 
 __attribute__((visibility("hidden")))
diff --git a/libsyscall/wrappers/mach_get_times.c b/libsyscall/wrappers/mach_get_times.c
index b078c8e..1e0fde2 100644
--- a/libsyscall/wrappers/mach_get_times.c
+++ b/libsyscall/wrappers/mach_get_times.c
@@ -25,7 +25,7 @@
 #include <sys/time.h>
 #include <mach/mach_time.h>
 #include <sys/time.h>
-#include <machine/cpu_capabilities.h>
+#include <i386/cpu_capabilities.h>
 
 // From __commpage_gettimeofday.c
 extern int __commpage_gettimeofday_internal(struct timeval *tp, uint64_t *tbr_out);
diff --git a/libsyscall/wrappers/spawn/posix_spawn.c b/libsyscall/wrappers/spawn/posix_spawn.c
index 3a14979..c7947f4 100644
--- a/libsyscall/wrappers/spawn/posix_spawn.c
+++ b/libsyscall/wrappers/spawn/posix_spawn.c
@@ -28,7 +28,7 @@
 #define CONFIG_MEMORYSTATUS 1 // <rdar://problem/13604997>
 #include <sys/types.h> /* for user_size_t */
 #include <spawn.h>
-#include <spawn_private.h>
+#include "spawn_private.h"
 #include <sys/spawn_internal.h>
 #include <sys/process_policy.h>
 #include <stdlib.h>
diff --git a/libsyscall/xcodescripts/compile-syscalls.pl b/libsyscall/xcodescripts/compile-syscalls.pl
index f0c2691..c69230e 100755
--- a/libsyscall/xcodescripts/compile-syscalls.pl
+++ b/libsyscall/xcodescripts/compile-syscalls.pl
@@ -63,7 +63,7 @@ my @CFLAGS = (
 	"-x assembler-with-cpp",
 	"-c",
 	"-isysroot", $ENV{'SDKROOT'} || "/",
-	"-I".$ENV{"SDKROOT"}."/System/Library/Frameworks/System.framework/PrivateHeaders",
+	"-I".$ENV{"DEPENDENCIES_DIR"}."/System/Library/Frameworks/System.framework/Versions/B/PrivateHeaders",
 );
 
 chomp(my $LIBTOOL = `xcrun -sdk "$ENV{'SDKROOT'}" -find libtool`);
