diff --git a/makedefs/MakeInc.def b/makedefs/MakeInc.def
index 6f6583b..a94076c 100644
--- a/makedefs/MakeInc.def
+++ b/makedefs/MakeInc.def
@@ -120,6 +120,7 @@ WARNFLAGS_STD := \
 	-Wfloat-conversion \
 	-Wconstant-conversion \
 	-Wpointer-bool-conversion \
+	-Wno-cast-qual \
 	-Wno-covered-switch-default \
 	-Wno-disabled-macro-expansion \
 	-Wno-documentation-unknown-command \
@@ -132,13 +133,15 @@ WARNFLAGS_STD := \
 	-Wno-padded \
 	-Wno-partial-availability \
 	-Wno-shift-sign-overflow \
+	-Wno-strict-prototypes \
 	-Wno-switch-enum \
 	-Wno-undef \
 	-Wno-unused-macros \
 	-Wno-used-but-marked-unused \
 	-Wno-variadic-macros \
 	-Wno-vla \
-	-Wno-zero-length-array
+	-Wno-zero-length-array \
+	-Wno-zero-as-null-pointer-constant
 
 WARNFLAGS_STD := $(WARNFLAGS_STD) \
 	-Wno-unknown-warning-option \
