diff --git a/xcodeconfig/libfirehose_kernel.xcconfig b/xcodeconfig/libfirehose_kernel.xcconfig
index c572f80..1c567bd 100644
--- a/xcodeconfig/libfirehose_kernel.xcconfig
+++ b/xcodeconfig/libfirehose_kernel.xcconfig
@@ -27,7 +27,7 @@ GCC_PREPROCESSOR_DEFINITIONS = $(inherited) KERNEL=1 DISPATCH_USE_DTRACE=0
 OTHER_CFLAGS = -mkernel -nostdinc -Wno-packed
 // LLVM_LTO = YES
 PRIVATE_HEADERS_FOLDER_PATH = /usr/local/include/kernel/os
-HEADER_SEARCH_PATHS = $(PROJECT_DIR) $(SDKROOT)/System/Library/Frameworks/Kernel.framework/PrivateHeaders $(SDKROOT)/System/Library/Frameworks/Kernel.framework/Headers $(SDKROOT)/usr/local/include/os $(SDKROOT)/usr/local/include/firehose
+HEADER_SEARCH_PATHS = $(PROJECT_DIR) $(DEPENDENCIES_DIR)/System/Library/Frameworks/Kernel.framework/PrivateHeaders $(DEPENDENCIES_DIR)/System/Library/Frameworks/Kernel.framework/Headers $(DEPENDENCIES_DIR)/usr/local/include/os $(DEPENDENCIES_DIR)/usr/local/include/firehose
 
 COPY_HEADERS_RUN_UNIFDEF = YES
 COPY_HEADERS_UNIFDEF_FLAGS = -DKERNEL=1 -DOS_FIREHOSE_SPI=1 -DOS_VOUCHER_ACTIVITY_SPI_TYPES=1 -UOS_VOUCHER_ACTIVITY_SPI
