diff --git a/src/firehose/firehose_buffer.c b/src/firehose/firehose_buffer.c
index 3bb790c..9dca1c6 100644
--- a/src/firehose/firehose_buffer.c
+++ b/src/firehose/firehose_buffer.c
@@ -1172,7 +1172,7 @@ void
 __firehose_buffer_tracepoint_flush(firehose_tracepoint_t ft,
 		firehose_tracepoint_id_u ftid)
 {
-	return firehose_buffer_tracepoint_flush(kernel_firehose_buffer, ft, ftid);
+	firehose_buffer_tracepoint_flush(kernel_firehose_buffer, ft, ftid);
 }
 
 void
